File: app/src/main/java/com/mdkashif/alarm/connectivity/ServiceGenerator.java
package com.mdkashif.alarm.connectivity;

import com.mdkashif.alarm.utils.AppConstants;

import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by Kashif on 09-Feb-18.
 * Retrofit Service Generator
 */

public class ServiceGenerator {

    private static Retrofit retrofit = null;

    public static Retrofit getClient() {

        if (retrofit==null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(AppConstants.PRAYER_API)
                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }
}

File: app/src/main/java/com/mdkashif/universalarm/connectivity/ServiceGenerator.kt
package com.mdkashif.universalarm.connectivity

import com.mdkashif.universalarm.utils.AppConstants

import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

/**
 * Created by Kashif on 09-Feb-18.
 * Retrofit Service Generator
 */

object ServiceGenerator {

    private var retrofit: Retrofit? = null

    val client: Retrofit
        get() {
            if (retrofit == null) {
                retrofit = Retrofit.Builder()
                        .baseUrl(AppConstants.GET_PRAYER_DATA)
                        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                        .addConverterFactory(GsonConverterFactory.create())
                        .build()
            }
            return retrofit!!
        }
}
